src/backend/data/mot/mot-object.cpp

```diff
@@ -44,7 +44,7 @@ int32_t pointer = 7;
 
 	headerSize     =
              ((segment [3] & 0x0F) << 9) |
-	               (segment [4] << 1) | ((segment [5] >> 7) && 0x01);
```

Comment:
The clang/clazy-driven automatic conversion of "&& 0x01" 'to true', and in effect the removal of this code is wrong. 

The code should probably be "& 0x01" (corrected in a later commit).